docs/platform_ccl.rst

```diff
@@ -0,0 +1,297 @@
+			{ "id": 3,  "x": 2, "y": 1 },
+			{ "id": 4,  "x": 4, "y": 1 },
+			{ "id": 5,  "x": 1, "y": 0 },
+			{ "id": 6,  "x": 3, "y": 0 }
+		],
+		"edges":
+		[
+			{ "id": 0,  "src": 2, "dst": 0 },
+			{ "id": 1,  "src": 0, "dst": 3 },
+			{ "id": 2,  "src": 3, "dst": 1 },
+			{ "id": 3,  "src": 1, "dst": 4 },
+			{ "id": 4,  "src": 2, "dst": 5 },
+			{ "id": 5,  "src": 5, "dst": 3 },
+			{ "id": 6,  "src": 3, "dst": 6 },
+			{ "id": 7,  "src": 6, "dst": 4 },
+			{ "id": 8,  "src": 0, "dst": 2 },
+			{ "id": 9,  "src": 3, "dst": 0 },
+			{ "id": 10,  "src": 1, "dst": 3 },
+			{ "id": 11,  "src": 4, "dst": 1 },
+			{ "id": 12,  "src": 5, "dst": 2 },
+			{ "id": 13,  "src": 3, "dst": 5 },
+			{ "id": 14,  "src": 6, "dst": 3 },
+			{ "id": 15,  "src": 4, "dst": 6 }
+		]
+	},
+
+
+Instructions can be specified/configured in ``instructions section``. An example of a 1-qubit 2-qubit instruction is shown below:
+
+.. code-block::
+   :linenos:
+
+   "instructions": {
+
+		"x q0": {
+			"duration": 40,
+			"latency": 0,
+			"qubits": ["q0"],
+			"matrix": [ [0.0,0.0], [1.0,0.0], 
+					    [1.0,0.0], [0.0,0.0]
+					  ],
+			"disable_optimization": false,
+			"type": "mw",
+			"cc_light_instr_type": "single_qubit_gate",
+			"cc_light_instr": "x",
+			"cc_light_codeword": 60,
+			"cc_light_opcode": 6
+		},
+
+		"cnot q2,q0": {
+			"duration": 80,
+			"latency": 0,
+			"qubits": ["q2","q0"],
+			"matrix": [ [0.1,0.0], [0.0,0.0], [0.0,0.0], [0.0,0.0],
+						[0.0,0.0], [1.0,0.0], [0.0,0.0], [0.0,0.0], 
+						[0.0,0.0], [0.0,0.0], [0.0,0.0], [1.0,0.0], 
+						[0.0,0.0], [0.0,0.0], [1.0,0.0], [0.0,0.0], 
+					  ],
+			"disable_optimization": true,
+			"type": "flux",
+			"cc_light_instr_type": "two_qubit_gate",
+			"cc_light_instr": "cnot",
+			"cc_light_right_codeword": 127,
+			"cc_light_left_codeword": 135,
+			"cc_light_opcode": 128
+	   },
+
+	   ...
+   }
+
+``x q0`` is the name of the instruction which will be used to refer to this instruction inside OpenQL program. The ``duration`` specifies the time duration required to complete this instruction. Due to control electronics, it is sometimes required to add a positive or negtive latency to an instruction. This can be specified by ``latency`` field. ``qubits`` refer to the list of qubits operands. ``matrix`` specifies the matrix representing this instruction. ``disable_optimization`` is used to enable disable optimization of instruction. An instruction can be of microwave, flux or readout type which is specified by the ``type`` field. ``cc_light_instr_type`` field is used to specify the type of instruction based on number of qubits. ``cc_light_instr`` specifies the name of this instruction used in CC-Light architecture. This name will be used in the generated output code. ``cc_light_right_codeword`` and ``cc_light_left_codeword`` is used to specify the codewords used for the left and right operation in CC-Light architecture. ``cc_light_opcode`` specifies the opcode used for this instruction.
```

Comment:
How will adding/removing latency affect the scheduling of a circuit? 
Will it simply attempt to move them earlier/later in the schedule? What if latency is not an even multiple of 20ns, will it round up or down?

Reply:
updated